bin/extract_galaxy_tools.py

```diff
@@ -336,59 +322,112 @@ def parse_tools(repo: Repository, ts_cat, excluded_tools, keep_tools):
-    parser.add_argument(
+    parser = argparse.ArgumentParser(
+        description="Extract Galaxy tools from GitHub repositories together with biotools and conda metadata"
+    )
+    subparser = parser.add_subparsers(dest="command")
+    # Extract tools
+    extractools = subparser.add_parser("extractools", help="Extract tools")
+    extractools.add_argument("--api", "-a", required=True, help="GitHub access token")
+    extractools.add_argument("--all_tools", "-o", required=True, help="Filepath to TSV with all extracted tools")
+    # Filter tools
+    filtertools = subparser.add_parser("filtertools", help="Filter tools")
+    filtertools.add_argument(
+        "--tools",
+        "-t",
+        required=True,
+        help="Filepath to TSV with all extracted tools, generated by extractools command",
+    )
+    filtertools.add_argument("--filtered_tools", "-f", required=True, help="Filepath to TSV with filtered tools")
+    filtertools.add_argument(
         "--categories", "-c", help="Path to a file with ToolShed category to keep in the extraction (one per line)"
     )
-    parser.add_argument("--exclude", "-e", help="Path to a file with ToolShed ids of tools to exclude (one per line)")
-    parser.add_argument("--keep", "-ek", help="Path to a file with ToolShed ids of tools to keep (one per line)")
+    filtertools.add_argument(
+        "--exclude", "-e", help="Path to a file with ToolShed ids of tools to exclude (one per line)"
+    )
+    filtertools.add_argument("--keep", "-k", help="Path to a file with ToolShed ids of tools to keep (one per line)")
     args = parser.parse_args()
 
-    # connect to GitHub
-    g = Github(args.api)
-    # get list of GitHub repositories to parse
-    repo_list = get_tool_github_repositories(g)
-
-    # get categories and tools to exclude
-    categories = read_file(args.categories)
-    excl_tools = read_file(args.exclude)
-    keep_tools = read_file(args.keep)
-
-    # parse tools in GitHub repositories to extract metada, filter by TS categories and export to output file
-    tools = []
-    for r in repo_list:
-        print(r)
-        if "github" not in r:
-            continue
-        repo = get_github_repo(r, g)
-        tools += parse_tools(repo, categories, excl_tools, keep_tools)
-        export_tools(tools, args.output)
-        print()
+    if args.command == "extractools":
+        # connect to GitHub
+        g = Github(args.api)
+        # get list of GitHub repositories to parse
+        repo_list = get_tool_github_repositories(g)
+        # parse tools in GitHub repositories to extract metada, filter by TS categories and export to output file
+        tools: List[Dict] = []
+        for r in repo_list:
+            print(r)
+            if "github" not in r:
+                continue
+            repo = get_github_repo(r, g)
+            tools += parse_tools(repo)
+            export_tools(tools, args.all_tools, format_list_col=True)
+            print()
```

Comment:
I suppose this should go outside of the for loop?

```suggestion
        export_tools(tools, args.all_tools, format_list_col=True)
```

Reply:
I put in the loop in case the job stops and we have some data already

Reply:
Ah, then I'll push a patch to write to a file handle instead.